tools/checkpoint_conversion/convert_phi3_checkpoints.py

```diff
@@ -0,0 +1,366 @@
+        "use_cache": False,
+        "output_attentions": False,
+        "output_hidden_states": False,
+        "return_dict": False,
+    }
+
+    hf_model_outputs = hf_model(**hf_model_input)[0]
+
+    # KerasNLP
+    keras_model_input = {
+        "token_ids": input_ids.to(keras_device),
+        "padding_mask": padding_mask.to(keras_device),
+    }
+    keras_model_outputs = keras_model(keras_model_input)
+
+    # Comparing the outputs.
+    print("🔶 KerasNLP output:", keras_model_outputs[0, 0, :10])
+    print("🔶 HF output:", hf_model_outputs[0, 0, :10])
+    print(
+        "🔶 Difference:",
+        torch.mean(
+            keras_model_outputs.detach().cpu() - hf_model_outputs.detach().cpu()
+        ),
+    )
+
+
+def get_torch_dtype(str_dtype):
+    if str_dtype == "float32":
+        return torch.float32
+    elif str_dtype == "float16":
+        return torch.float16
+    elif str_dtype == "bfloat16":
+        return torch.bfloat16
+
+
+def convert_and_validate(
+    hf_model_dir,
+    hf_device,
+    keras_device,
+    validate_dtype,
+):
+    print(f"✅ Numerics Validatiion in {validate_dtype}.")
+    # Load the causal model to convert lm_head weights.
+    hf_causal_model = transformers.AutoModelForCausalLM.from_pretrained(
+        hf_model_dir,
+        device_map=hf_device,
+        torch_dtype=get_torch_dtype(validate_dtype),
+        trust_remote_code=True,
+    )
+    hf_model = hf_causal_model.model
+    print("✅ Huggingface model loaded.")
+
+    keras_model = convert_model(hf_causal_model, keras_device, validate_dtype)
+    print("✅ Keras model loaded.")
+
+    convert_weights(keras_model, hf_causal_model)
+    print("✅ Weights converted")
+
+    validate_output(
+        hf_model,
+        keras_model,
+        hf_device,
+        keras_device,
+    )
+    print("✅ Numerics validated")
+
+    # Clean memory.
+    del keras_model
+    del hf_causal_model
+    del hf_model
+    gc.collect()
+    if not (hf_device == "cpu" and keras_device == "cpu"):
+        torch.cuda.empty_cache()
+
+
+def convert_and_save(
+    preset,
+    hf_model_dir,
+    hf_device,
+    keras_device,
+    save_dtype,
+):
+    print(f"✅ Saving model in {save_dtype}.")
+    # Load the causal model to convert lm_head weights.
+    hf_causal_model = transformers.AutoModelForCausalLM.from_pretrained(
+        hf_model_dir,
+        device_map=hf_device,
+        torch_dtype=get_torch_dtype(save_dtype),
+        trust_remote_code=True,
+    )
+    hf_model = hf_causal_model.model
+    print("✅ Huggingface model loaded.")
+
+    keras_model = convert_model(hf_causal_model, keras_device, save_dtype)
+    print("✅ Keras model loaded.")
+
+    convert_weights(keras_model, hf_causal_model)
+    print("✅ Weights converted")
+
+    validate_output(
+        hf_model,
+        keras_model,
+        hf_device,
+        keras_device,
+    )
+    print("✅ Numerics validated")
+
+    keras_model.save_to_preset(preset)
+    print("✅ Preset saved")
```

Comment:
BTW, now you can also upload the model presets to Kaggle or Huggingface directly using the `keras_nlp.upload_preset` function. For Kaggle, you can create a model page (public or private) and upload using the URI: 

```python
import keras_nlp

preset = "phi3_mini_4k_instruct_en"
model = keras_nlp.Phi3Backbone(...)
model.save_to_preset(f"{preset}")
keras_nlp.upload_preset(f"kaggle://{username}/{model_page_name}/keras/{preset}")
```

Might be useful for write tests for the presets. But also, no need to do that just yet, we can do that in the Causal LM PR instead.

Reply:
Thanks I will add add it to the conversion script when I add the tokenizer.